src/connect/ConnectWizard.ts

```diff
@@ -51,93 +52,34 @@ export class ConnectWizard {
-                    }
-                    return this.pickServiceAsync(input, currentContext, targetResourceType);
-                }
-
-                // A target resource was passed in, so we can can skip some configuration steps.
-                // Validate the current context against the selected resource
-                if (targetResourceNamespace != null && currentContext.namespace != null && currentContext.namespace !== targetResourceNamespace) {
-                    throw new Error(`The ${targetResourceType} '${targetResourceName}' belongs to the namespace '${targetResourceNamespace}', `
-                                    + `but the current context targets namespace '${currentContext.namespace}'. Please update your kubeconfig to target the correct context.`);
-                }
-
-                let namespaces: string[] = null;
-                try {
-                    namespaces = await kubectlClient.getNamespacesAsync(currentContext.kubeconfigPath);
-                }
-                catch (error) {
-                    // We want to recover if for some reason the user isn't able to list namespaces.
-                    this._logger.warning(`Failed to list namespaces`, error);
-                }
-                if (namespaces != null && !namespaces.includes(targetResourceNamespace)) {
-                    // In practice, this error should happen rarely, as the K8s cluster explorer only allows interaction with resources from the current cluster
-                    throw new Error(`Failed to find the namespace '${targetResourceNamespace}' in cluster '${currentContext.cluster}'`);
-                }
-
-                let resourceNameToPersist: string = targetResourceName;
-                if (targetResourceType === ResourceType.Pod) {
-                    // Because the specific pod name changes, we persist only the first segments of the name, e.g. <deployment name>-<deployment guid>-.
-                    const split: string[] = targetResourceName.split(`-`);
-                    if (split.length > 1) {
-                        // Delete last segment
-                        split.splice(-1, 1);
-                        resourceNameToPersist = `${split.join(`-`)}`;
-                    }
-                }
-
-                this._result.resourceName = resourceNameToPersist;
-                this._result.targetCluster = currentContext.cluster;
-                this._result.targetNamespace = targetResourceNamespace;
-                this._result.resourceType = targetResourceType;
-
-                return (input: MultiStepInput): Promise<(input: MultiStepInput) => Promise<(input: MultiStepInput) => Promise<void>>> => this.inputPortsAsync(input, targetResourceType);
-            });
-        }
-        catch (error) {
+                    throw new Error(`Target resource name cannot be unset for resource type ${targetResourceType}`);
```

Comment:
It looks as if this path was previously hit when `targetResourceName === null && targetResourceType === 'pod'`. So I think this check should be moved inside `handlePodResourceType`.

It looks as if the code never handled the case where `targetResourceType` was something other than `pod` or `service`. Maybe that's handled by the caller of this function?

So, I'd still throw an error here, but maybe change the comment and the error message (`Unexpected resource type ${targetResourceType}`?